ui/core/src/api/utils/parseConfig.ts

```diff
@@ -0,0 +1,63 @@
+import { ApiContext } from "..";
+import { Coin, Network, Token } from "../../entities";
+import { getMetamaskProvider } from "../EthereumService/utils/getMetamaskProvider";
+
+type TokenConfig = {
+  symbol: string;
+  decimals: number;
+  imageUrl?: string;
+  name: string;
+  address: string;
+  network: Network;
+};
+
+type CoinConfig = {
+  symbol: string;
+  decimals: number;
+  imageUrl?: string;
+  name: string;
+  network: Network;
+};
+
+type AssetConfig = CoinConfig | TokenConfig;
+function isTokenConfig(a: any): a is TokenConfig {
+  return typeof a?.address === "string";
+}
+
+function parseAsset(a: unknown) {
+  if (isTokenConfig(a)) {
+    return Token(a);
+  }
+  return Coin(a as CoinConfig);
+}
+
+export type ChainConfig = {
```

Comment:
I could see us wanting ChainConfig separate from providers, assets, etc

Reply:
I woke up and had exactly the same thought. I was thinking I could knock that up this morning before the meeting but that we should get this in so develop isn’t blocked